code/modules/materials/_materials.dm

```diff
@@ -385,7 +388,25 @@
 	holder.remove_reagent(type, removed)
 
 /decl/material/proc/affect_blood(var/mob/living/carbon/M, var/alien, var/removed, var/datum/reagents/holder)
-	return
+
+	if(toxicity)
+		M.add_chemical_effect(CE_TOXIN, toxicity)
+		var/dam = (toxicity * removed)
+		if(toxicity_targets_organ && ishuman(M))
+			var/mob/living/carbon/human/H = M
+			var/obj/item/organ/internal/I = H.internal_organs_by_name[toxicity_targets_organ]
+			if(I)
+				var/can_damage = I.max_damage - I.damage
+				if(can_damage > 0)
+					if(dam > can_damage)
+						I.take_internal_damage(can_damage, silent=TRUE)
+						dam -= can_damage
+					else
+						I.take_internal_damage(dam, silent=TRUE)
+						dam = 0
+		if(dam)
```

Comment:
Usually best to do `if(dam > 0)` or similar to avoid issues with float arithmetic.